C error when trying to check rune values

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: windows, Microsoft Windows 10 Home v19043 64-bit
Processor: 4 cpus, 64bit, little endian, Intel(R) Celeron(R) N4120 CPU @ 1.10GHz  

CC version: Error: exec failed (CreateProcess) with code 2: The system cannot find the file specified.

 cmd: cc --version

getwd: C:\Users\user\workspace\src\Zen
vmodules: C:\Users\user\.vmodules
vroot: C:\Program Files\VLang\v
vexe: C:\Program Files\VLang\v\v.exe
vexe mtime: 2022-04-17 17:35:20
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 4f14f77.56a3539

Git version: git version 2.34.1.windows.1
Git vroot status: weekly.2022.15-68-g56a3539e
.git/config present: true
thirdparty/tcc status: thirdparty-windows-amd64 9de1a49e

```

**What did you do?**
`v src/language_v/. -o zen.exe`
```v
// This is the file that has failed compilation
module scanner


struct Scanner<T>
{
	contents []T
}


pub fn (s Scanner<T>) tokenize()
{
	for i, c in s.contents
	{
		if c == `\n`
		{
			println('$i: \\n')
		}

		else if c == '\r'
		{
			println('$i: \\r')
		}

		else if c == '\t'
		{
			println('$i: \\t')
		}

		else if c == ' '
		{
			println('$i: " "')
		}

		else
		{
			println('$i: $c')
		}
	}
}


pub fn new_scanner<T>(contents []T) Scanner<T>
{
	return Scanner<T>{contents}
}

```

**What did you expect to see?**

A printed string with special characters such as \\n and \\t to be printed in their raw format


**What did you see instead?**
```
==================
C:/Users/user/AppData/Local/Temp/v_0/zen.exe.16067074313486762462.tmp.c:7849: warning: cast between pointer and integer of different size
C:/Users/user/AppData/Local/Temp/v_0/zen.exe.16067074313486762462.tmp.c:12466: error: invalid operand types for binary operation
C:/Users/user/AppData/Local/Temp/v_0/zen.exe.16067074313486762462.tmp.c:12468: error: invalid operand types for binary operation
C:/Users/user/AppData/Local/Temp/v_0/zen.exe.16067074313486762462.tmp.c:12470: error: invalid operand types for binary operation
C:/Users/user/AppData/Local/Temp/v_0/zen.exe.16067074313486762462.tmp.c:13856: warning: assignment from incompatible pointer type
C:/Users/user/AppData/Local/Temp/v_0/zen.exe.16067074313486762462.tmp.c:13947: warning: implicit declaration of function 'GetFinalPathNameByHandleW'
C:/Users/user/AppData/Local/Temp/v_0/zen.exe.16067074313486762462.tmp.c:15580: warning: implicit declaration of function 'CreateSymbolicLinkW'
C:/Users/user/AppData/Local/Temp/v_0/zen.exe.16067074313486762462.tmp.c:15640: warning: assignment from incompatible pointer type
C:/Users/user/AppData/Local/Temp/v_0/zen.exe.16067074313486762462.tmp.c:15652: warning: assignment from incompatible pointer type
C:/Users/user/AppData/Local/Temp/v_0/zen.exe.16067074313486762462.tmp.c:16115: warning: assignment from incompatible pointer type
C:/Users/user/AppData/Local/Temp/v_0/zen.exe.16067074313486762462.tmp.c:16205: warning: assignment from incompatible pointer type
...
==================
(Use `v -cg` to print the entire error message)

builder error:
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

--- a/vlib/v/checker/infix.v
+++ b/vlib/v/checker/infix.v
@@ -734,6 +734,14 @@ fn (mut c Checker) infix_expr(mut node ast.InfixExpr) ast.Type {
 
 	// Dual sides check (compatibility check)
 	if node.left !is ast.ComptimeCall && node.right !is ast.ComptimeCall {
+		if left_type.has_flag(.generic) {
+			left_type = c.unwrap_generic(left_type)
+			left_sym = c.table.sym(left_type)
+		}
+		if right_type.has_flag(.generic) {
+			right_type = c.unwrap_generic(right_type)
+			right_sym = c.table.sym(right_type)
+		}
 		if !(c.symmetric_check(left_type, right_type) && c.symmetric_check(right_type, left_type))
 			&& !c.pref.translated && !c.file.is_translated && !node.left.is_auto_deref_var()
 			&& !node.right.is_auto_deref_var() {

--- a/vlib/v/checker/tests/generic_eq_wrong_type.err.out
+++ b/vlib/v/checker/tests/generic_eq_wrong_type.err.out
@@ -1,0 +1,7 @@
+vlib/v/checker/tests/generic_eq_wrong_type.err.vv:9:13: error: infix expr: cannot use `string` (right expression) as `rune`
+    7 |         if ztest == `\n` {
+    8 |             println('${z}: \\n')
+    9 |         } else if ztest == '\r' {
+      |                   ~~~~~~~~~~~~~
+   10 |             println('${z}: \\r')
+   11 |         } else {

--- a/vlib/v/checker/tests/generic_eq_wrong_type.err.vv
+++ b/vlib/v/checker/tests/generic_eq_wrong_type.err.vv
@@ -1,0 +1,24 @@
+struct Scanner[T] {
+	contents []T
+}
+
+pub fn (s Scanner[T]) tokenize() {
+	for z, ztest in s.contents {
+		if ztest == `\n` {
+			println('${z}: \\n')
+		} else if ztest == '\r' {
+			println('${z}: \\r')
+		} else {
+			println('${z}: ${ztest}')
+		}
+	}
+}
+
+pub fn new_scanner[T](contents []T) Scanner[T] {
+	return Scanner[T]{contents}
+}
+
+fn main() {
+	s := new_scanner([`\n`, `b`])
+	s.tokenize()
+}

